sql/catalyst/src/main/java/org/apache/spark/sql/connector/util/V2ExpressionSQLBuilder.java

```diff
@@ -87,14 +96,21 @@ public String build(Expression expr) {
     }
   }
 
-  protected String visitLiteral(LiteralValue literalValue) {
-    return literalValue.toString();
+  protected String visitLiteral(Literal literal) {
+    return literal.toString();
   }
 
   protected String visitFieldReference(FieldReference fieldRef) {
     return fieldRef.toString();
   }
 
+  protected String visitIn(String v, List<String> list) {
+    if (list.isEmpty()) {
+      return "CASE WHEN " + v + " IS NULL THEN NULL ELSE FALSE END";
```

Comment:
isn't it just `IF(v IS NULL, NULL, FALSE)`

Reply:
IF is not a SQL standard. Many databases doesn't support it.